crates/ibc/src/core/ics04_channel/handler/recv_packet.rs

```diff
@@ -52,87 +52,108 @@ pub fn process<Ctx: ChannelReader>(
+    let conn_end_on_b = ctx_b
+        .connection_end(conn_id_on_b)
         .map_err(PacketError::Channel)?;
 
-    if !connection_end.state_matches(&ConnectionState::Open) {
+    if !conn_end_on_b.state_matches(&ConnectionState::Open) {
         return Err(PacketError::ConnectionNotOpen {
-            connection_id: dest_channel_end.connection_hops()[0].clone(),
+            connection_id: chan_end_on_b.connection_hops()[0].clone(),
         });
     }
 
-    let latest_height = ChannelReader::host_height(ctx).map_err(PacketError::Channel)?;
+    let latest_height = ChannelReader::host_height(ctx_b).map_err(PacketError::Channel)?;
     if packet.timeout_height.has_expired(latest_height) {
         return Err(PacketError::LowPacketHeight {
             chain_height: latest_height,
             timeout_height: packet.timeout_height,
         });
     }
 
-    let latest_timestamp = ChannelReader::host_timestamp(ctx).map_err(PacketError::Channel)?;
+    let latest_timestamp = ChannelReader::host_timestamp(ctx_b).map_err(PacketError::Channel)?;
     if let Expiry::Expired = latest_timestamp.check_expiry(&packet.timeout_timestamp) {
         return Err(PacketError::LowPacketTimestamp);
     }
 
-    verify_packet_recv_proofs(
-        ctx,
-        msg.proofs.height(),
-        packet,
-        &connection_end,
-        &msg.proofs,
-    )
-    .map_err(PacketError::Channel)?;
+    // Verify proofs
+    {
+        let client_id_on_b = conn_end_on_b.client_id();
+        let client_state_of_a_on_b = ctx_b
+            .client_state(client_id_on_b)
+            .map_err(PacketError::Channel)?;
+
+        // The client must not be frozen.
+        if client_state_of_a_on_b.is_frozen() {
+            return Err(PacketError::FrozenClient {
+                client_id: client_id_on_b.clone(),
+            });
+        }
 
-    let result = if dest_channel_end.order_matches(&Order::Ordered) {
-        let next_seq_recv =
-            ctx.get_next_sequence_recv(&packet.destination_port, &packet.destination_channel)?;
+        let consensus_state_of_a_on_b = ctx_b
+            .client_consensus_state(client_id_on_b, &msg.proofs.height())
+            .map_err(PacketError::Channel)?;
 
-        if packet.sequence < next_seq_recv {
-            output.emit(IbcEvent::ReceivePacket(ReceivePacket::new(
-                msg.packet.clone(),
-                dest_channel_end.ordering,
-                dest_connection_id.clone(),
-            )));
-
-            return Ok(output.with_result(PacketResult::Recv(RecvPacketResult::NoOp)));
-        } else if packet.sequence != next_seq_recv {
+        let commitment = ctx_b.packet_commitment(
+            &packet.data,
+            &packet.timeout_height,
+            &packet.timeout_timestamp,
+        );
+        // Verify the proof for the packet against the chain store.
+        client_state_of_a_on_b
+            .verify_packet_data(
+                ctx_b,
+                msg.proofs.height(),
```

Comment:
The only thing left to do is to change the message structs to completely remove the use of `Proofs`. You can look at `MsgConnectionOpenTry` for an example of how to do that (and/or the diff in PR #170).

Reply:
Should be fine now [65ee023](https://github.com/cosmos/ibc-rs/pull/323/commits/65ee023c880f928b690583c0496b22b31ce646cf)